demo/kserve/custom-manifests/caikit/caikit-tgis-isvc-template.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  annotations:
+    serving.knative.openshift.io/enablePassthrough: "true"
+    sidecar.istio.io/inject: "true"
+    sidecar.istio.io/rewriteAppHTTPProbers: "true"
+  name: caikit-tgis-isvc
+spec:
+  predictor:
+    serviceAccountName: sa
+    model:
+      modelFormat:
+        name: caikit
+      runtime: caikit-tgis-runtime
+      storageUri: proto://path/to/model # single model here
+      # Example, using a pvc:
+      # storageUri: pvc://caikit-pvc/flan-t5-small-caikit/
+      # Target directory must contain a config.yml
```

Comment:
Do we really need this?

Reply:
The comment above by @Jooho asked to keep the generic YAML so that it should be adapted to sources other than the example's minio. In response, we kept this file (only added the missing `sa`) but also provided a concrete yaml that works  with the overall scenario's `minio`.

Reply:
this yaml is using annotation way that kserve originally used but ODH dashboard only use json style `storage-config`  [example](https://github.com/kserve/modelmesh-serving/blob/bba0cec8ca8c6c6f19958696f39b27b5b49cadd8/docs/predictors/setup-storage.md#3-add-a-storage-entry-to-the-storage-config-secret)
With json style, we need to use key/path instead of storageUri. 
```
  predictor:
    model:
      modelFormat:
        name: tensorflow
        version: "1.15" # Optional
      runtime: triton-2.x #Optional
      storage:
        key: my_storage
        path: my_models/mnist-tf
```
In this way, we don't need ServiceAccount anymore.

Reply:
I understand the SA is not needed because the storage access is defined inside the ISVC. I also understand the example, which is about defining a storage secret as a JSON construct. So in summary, we need to implement these changes:
1. Define the `storage-config` secret with JSON
2. Add the storage URI and storage secret key directly inside the ISVC
3. Drop the SA - not needed anymore 
Agreed?